src/NuGet.Core/NuGet.Protocol.VisualStudio/Resources/ISearchResultsIndexer.cs

```diff
@@ -0,0 +1,10 @@
+﻿using NuGet.Protocol.Core.Types;
+using System.Collections.Generic;
+
+namespace NuGet.Protocol.VisualStudio
+{
+    public interface ISearchResultsIndexer : INuGetResource
+    {
+        IDictionary<string, int> Rank(string queryString, IEnumerable<PackageSearchMetadata> entries);
```

Comment:
This is arguable. But I'd prefer to postpone this discussion for now. Will make it a regular class and push as a functional component first.